src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -468,6 +469,28 @@ static int open_device(const char* path, bool try_load_moudle = false)
   return nbd;
 }
 
+static int check_device_size(int nbd_index, unsigned long expected_size)
+{
+  unsigned long size = 0;
```

Comment:
I suggest use uint64_t instead of unsigned long here. size *= 512 may overflow...